wasm/jit/jit_amd64_test.go

```diff
@@ -1402,6 +1402,97 @@ func TestAmd64Builder_handleCall(t *testing.T) {
 	})
 }
 
+func TestAmd64Builder_handleLoad(t *testing.T) {
+	for i, tp := range []wazeroir.SignLessType{
+		wazeroir.SignLessTypeI32,
+		wazeroir.SignLessTypeI64,
+		wazeroir.SignLessTypeF32,
+		wazeroir.SignLessTypeF64,
+	} {
+		tp := tp
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			eng := newEngine()
+			builder := requireNewBuilder(t)
+			builder.initializeReservedRegisters()
+			// Before load operations, we must push the base offset value.
+			const baseOffset = 100
+			base := builder.locationStack.pushValueOnStack()
+			eng.stack[base.stackPointer] = baseOffset
+			// Emit the memory load instructions.
+			o := &wazeroir.OperationLoad{Type: tp, Arg: &wazeroir.MemoryImmediate{Offest: 361}}
+			err := builder.handleLoad(o)
+			require.NoError(t, err)
+			// At this point, the loaded value must be on top of the stack, and placed on a register.
+			loadedValue := builder.locationStack.peek()
+			require.Equal(t, o.Type, loadedValue.valueType)
+			require.True(t, loadedValue.onRegister())
+			// Double the laoded value in order to verify the behavior.
+			var addInst obj.As
+			switch tp {
+			case wazeroir.SignLessTypeI32:
+				require.True(t, isIntRegister(loadedValue.register))
+				addInst = x86.AADDL
+			case wazeroir.SignLessTypeI64:
+				require.True(t, isIntRegister(loadedValue.register))
+				addInst = x86.AADDQ
+			case wazeroir.SignLessTypeF32:
+				require.True(t, isFloatRegister(loadedValue.register))
+				addInst = x86.AADDSS
+			case wazeroir.SignLessTypeF64:
+				require.True(t, isFloatRegister(loadedValue.register))
+				addInst = x86.AADDSD
+			}
+			doubleLoadedValue := builder.newProg()
+			doubleLoadedValue.As = addInst
+			doubleLoadedValue.To.Type = obj.TYPE_REG
+			doubleLoadedValue.To.Reg = loadedValue.register
+			doubleLoadedValue.From.Type = obj.TYPE_REG
+			doubleLoadedValue.From.Reg = loadedValue.register
+			builder.addInstruction(doubleLoadedValue)
+
+			// Compile.
+			builder.releaseAllRegistersToStack()
+			builder.returnFunction()
```

Comment:
why we are doing releaseAllRegistersToStack and returnFunction is special knowledge unless you comment why :D